src/oci.rs

```diff
@@ -142,6 +147,75 @@ fn init_dir(layout: impl AsRef<Path>) -> Result<(), anyhow::Error> {
     Ok(())
 }
 
+// https://mgorny.pl/articles/portability-of-tar-features.html#id25
+const PAX_SCHILY_XATTR: &[u8; 13] = b"SCHILY.xattr.";
+
+// workaround https://github.com/alexcrichton/tar-rs/issues/102 so that security capabilities are preserved
+fn append_dir_all_with_xattrs(
+    tar: &mut tar::Builder<impl Write>,
+    src_path: impl AsRef<Path>,
+) -> Result<()> {
+    let src_path = src_path.as_ref();
+    for entry in WalkDir::new(src_path)
+        .follow_links(false)
+        .into_iter()
+        .filter_map(Result::ok)
+    {
+        let meta = entry.metadata()?;
+        // skip sockets as tar-rs errors when trying to archive them.
+        // For comparison, umoci also errors, whereas docker skips them
+        if meta.file_type().is_socket() {
+            continue;
+        }
+
+        let rel_path = pathdiff::diff_paths(entry.path(), src_path).unwrap();
```

Comment:
entry.path() should always be under src_path but if not then this unwrap will panic. 
Could you error cleanly in that case.

Reply:
I've added an expect clarifying the assumption I'm making here.